src/datasets/load.py

```diff
@@ -1525,15 +1694,15 @@ def load_dataset_builder(
     # Instantiate the dataset builder
     builder_instance: DatasetBuilder = builder_cls(
         cache_dir=cache_dir,
+        dataset_name=dataset_name,
         config_name=config_name,
-        data_dir=data_dir,
-        data_files=data_files,
+        data_dir=data_dir if not dataset_module.metadata_configs else None,
+        data_files=data_files if not dataset_module.metadata_configs else None,
```

Comment:
I think we can keep data_dir and data_files no ? In case someone wants to load a subset manually ?

The config_id would change so we won't have issues with caching